### PR TITLE
Fire propagating events only after elements are attached

### LINKED ISSUE #552: "frozen" doesn't get propagated to all cells inside a frozen column group (iOS9)
Works in iOS10

### PR DESCRIPTION
**Pull request by Saulis**
Fixes #552

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-grid/613)

### CODE REVIEW

File: vaadin-grid-table-cell.html [hunk 1]
```diff
@@ -246,7 +249,10 @@
           this.instance = {root: document.createElement('div')};
         }
 
-        this.fire('cell-empty-changed');
+        // Safari 9 doesn't bubble events while not attached to the DOM. #552
```
Comment (limonte) [inline vaadin-grid-table-cell.html]: Is this a Safari issue or an iOS issue?

**Comment by platosha** [comment]
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[test/column-group.html, line 118 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/613#-K_-MnOkNZLKtv1PC6UG:-K_-MnOlwU8icdE2EFjm:bt5u0s5) ([raw file](https://github.com/vaadin/vaadin-grid/blob/d7ef0e53d5e83d01225307562e70a8c54cd48cf5/test/column-group.html#L118)):*
> ```HTML
> 
>       // this test is aimed for Safari 9, see #552
>       it('should frozen is set on attach', function() {
> ```

Should do what? Please reformulate the test title.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/613)*

**Comment by manolo** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-grid-column.html, line 142 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/613#-K_0R9MeWkiPKzzY5vdn:-K_0R9MeWkiPKzzY5vdo:b-kxhnk0) ([raw file](https://github.com/vaadin/vaadin-grid/blob/d7ef0e53d5e83d01225307562e70a8c54cd48cf5/vaadin-grid-column.html#L142)):*
> ```HTML
>     ],
> 
>     _frozenChanged: function(frozen, isAttached) {
> ```

does this issue happen only with frozen? I guess  other properties could have the problem as well.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/613)*

**Comment by Saulis** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-grid-column.html, line 142 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/613#-K_0R9MeWkiPKzzY5vdn:-K_0eh_eJGxUDP55jdRX:bweqnuh) ([raw file](https://github.com/vaadin/vaadin-grid/blob/d7ef0e53d5e83d01225307562e70a8c54cd48cf5/vaadin-grid-column.html#L142)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

does this issue happen only with frozen? I guess  other properties could have the problem as well.
</blockquote></details>

This is so far the only column group related property we've encountered the issue. I tested out `hidden` but it does not have any issues. The only difference I see between `hidden` and `frozen` is that `frozen` is defined in the base behavior which makes `_frozenChanged` being triggered earlier than `_hiddenChanged` (don't know why)

---

*[vaadin-grid-table-cell.html, line 252 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/613#-KZwU8wcGBzucq_sIQ_U-r1-252:-K_0eSmKPfaDKq_kGh-I:b-5v8h5i) ([raw file](https://github.com/vaadin/vaadin-grid/blob/d7ef0e53d5e83d01225307562e70a8c54cd48cf5/vaadin-grid-table-cell.html#L252)):*
<details><summary><i>Previously, limonte (Limon Monte) wrote…</i></summary><blockquote>

Is this a Safari issue or an iOS issue?
</blockquote></details>

Yes. :smile: 

iOS 9 runs (mobile) Safari 9.

---

*[test/column-group.html, line 118 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/613#-K_-MnOkNZLKtv1PC6UG:-K_0eRj0wI5D57rb3tYl:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid/blob/d7ef0e53d5e83d01225307562e70a8c54cd48cf5/test/column-group.html#L118)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Should do what? Please reformulate the test title.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/613)*

**Comment by limonte** [comment]
Review status: 2 of 3 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[vaadin-grid-table-cell.html, line 252 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/613#-KZwU8wcGBzucq_sIQ_U-r1-252:-K_3PJDO-IAynrql4pMg:b76c1xh) ([raw file](https://github.com/vaadin/vaadin-grid/blob/d7ef0e53d5e83d01225307562e70a8c54cd48cf5/vaadin-grid-table-cell.html#L252)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

Yes. :smile: 

iOS 9 runs (mobile) Safari 9.
</blockquote></details>

- do you prefer blondes or brunettes
- yes.

:D

I'll rephrase - is there the same issue in iOS/Chrome?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/613)*

**Comment by limonte** [comment]
:lgtm:

---

Review status: 1 of 3 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/613#-:-K_3RGrOLenG7pbDFgwl:bnfp4nl)*

**Comment by platosha** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/613#-:-K_3hcba3TZ07DPWUx74:bnfp4nl)*

**Comment by Saulis** [comment]
Reviewed 1 of 3 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/613)*

**Comment by manolo** [comment]
:lgtm:

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/613#-:-K_8Nq0VLyZ2dlAua37m:bnfp4nl)*

**Pull request merged**